src/VideoCommand.cc

```diff
@@ -0,0 +1,352 @@
+                get_value<int>(op, "stop"),
+                get_value<int>(op, "step"));
+
+        }
+        else if (type == "resize") {
+             video.resize(get_value<int>(op, "height"),
+                           get_value<int>(op, "width") );
+
+        }
+        else if (type == "crop") {
+            video.crop(VCL::Rectangle (
+                        get_value<int>(op, "x"),
+                        get_value<int>(op, "y"),
+                        get_value<int>(op, "width"),
+                        get_value<int>(op, "height") ));
+        }
+        else {
+            throw ExceptionCommand(ImageError, "Operation not defined");
+        }
+    }
+}
+
+VCL::Video::Codec VideoCommand::string_to_codec(std::string& codec)
+{
+    if (codec == "h263") {
+        return VCL::Video::Codec::H263;
+    }
+    else if (codec == "xvid") {
+        return VCL::Video::Codec::XVID;
+    }
+    else if (codec == "h264") {
+        return VCL::Video::Codec::H264;
+    }
+
+    return VCL::Video::Codec::NOCODEC;
+}
+
+//========= AddVideo definitions =========
+
+AddVideo::AddVideo() : VideoCommand("AddVideo")
+{
+    _storage_video = VDMSConfig::instance()->get_path_videos();
+}
+
+int AddVideo::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    int node_ref = get_value<int>(cmd, "_ref",
+                                  query.get_available_reference());
+
+    VCL::Video video((void*)blob.data(), blob.size());
+
+    if (cmd.isMember("operations")) {
+        enqueue_operations(video, cmd["operations"]);
+    }
+
+    // The container and codec are checked by the schema.
+    // We default to mp4 and h264, if not specified
+    std::string container  = get_value<std::string>(cmd, "container", "mp4");
+    std::string file_name  = VCL::create_unique(_storage_video, container);
+
+    // Modifiyng the existing properties that the user gives
+    // is a good option to make the AddNode more simple.
+    // This is not ideal since we are manupulating with user's
+    // input, but for now it is an acceptable solution.
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
+    props[VDMS_VID_PATH_PROP] = file_name;
+
+    // Add Video node
+    query.AddNode(node_ref, VDMS_VID_TAG, props, Json::Value());
+
+    std::string codec = get_value<std::string>(cmd, "codec", "h264");
+    VCL::Video::Codec vcl_codec = string_to_codec(codec);
+
+    video.store(file_name, vcl_codec);
+
+    // In case we need to cleanup the query
+    error["video_added"] = file_name;
+
+    if (cmd.isMember("link")) {
+        add_link(query, cmd["link"], node_ref, VDMS_VID_EDGE);
+    }
+
+    return 0;
+}
+
+//========= UpdateImage definitions =========
+
+UpdateVideo::UpdateVideo() : VideoCommand("UpdateVideo")
+{
+}
+
+int UpdateVideo::construct_protobuf(
+    PMGDQuery& query,
```

Comment:
No update wiki page

Reply:
So its not clear what update will support updating for videos.

Reply:
this is just to keep it the same as image. For now, it just updates the properties (same as images)